janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -561,10 +558,17 @@ private void tokenize(SearchParams params, final Mapping mapping, final LuceneCu
                 params.addQuery(new PrefixQuery(term), BooleanClause.Occur.MUST);
             } else throw new IllegalArgumentException("LuceneIndex does not support this predicate with 1 token : " + janusgraphPredicate);
         } else {
-            // at the moment, this is only walked for EQUAL and Text.CONTAINS (String and Text mappings)
+            // at the moment, this is only walked for EQUAL, NOT_EQUAL and Text.CONTAINS (String and Text mappings)
             final BooleanQuery.Builder q = new BooleanQuery.Builder();
+            BooleanClause.Occur occur;
+            if (janusgraphPredicate == Cmp.NOT_EQUAL) {
+                q.add(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
+                occur = BooleanClause.Occur.MUST_NOT;
```

Comment:
Could you please add a test for this part?
Just to check whether it's tested, I changed this to `MUST` instead of `MUST_NOT` and the test still passed.

Reply:
Ok, I'm add a test that cover this case